pkg/controller/storagecluster/externalstoragecluster_controller.go

```diff
@@ -0,0 +1,245 @@
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// ReconcileExternalStorageCluster reconciles the external Storage Cluster
+func (r *ReconcileStorageCluster) ReconcileExternalStorageCluster(sc *ocsv1.StorageCluster) (reconcile.Result, error) {
+	reqLogger := r.reqLogger.WithValues("Request.Namespace", sc.Namespace, "Request.Name", sc.Name)
+	reqLogger.Info("Reconciling External StorageCluster")
+	var err error
+
+	// Add conditions if there are none
+	if sc.Status.Conditions == nil {
+		reason := ocsv1.ReconcileInit
+		message := "Initializing External StorageCluster"
+		statusutil.SetProgressingCondition(&sc.Status.Conditions, reason, message)
+		err = r.client.Status().Update(context.TODO(), sc)
+		if err != nil {
+			reqLogger.Error(err, "Failed to add conditions to status")
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Check GetDeletionTimestamp to determine if the object is under deletion
+	if sc.GetDeletionTimestamp().IsZero() {
+		if !contains(sc.GetFinalizers(), storageClusterFinalizer) {
+			reqLogger.Info("Finalizer not found for storagecluster. Adding finalizer")
+			sc.ObjectMeta.Finalizers = append(sc.ObjectMeta.Finalizers, storageClusterFinalizer)
+			if err := r.client.Update(context.TODO(), sc); err != nil {
+				reqLogger.Error(err, "Failed to update storagecluster with finalizer")
+				return reconcile.Result{}, err
+			}
+		}
+	} else {
+		// The object is marked for deletion
+		sc.Status.Phase = statusutil.PhaseDeleting
+		phaseErr := r.client.Status().Update(context.TODO(), sc)
+		if phaseErr != nil {
+			reqLogger.Error(phaseErr, "Failed to set PhaseDeleting")
+		}
+		if contains(sc.GetFinalizers(), storageClusterFinalizer) {
+			isDeleted, err := r.deleteResources(sc, reqLogger)
+			if err != nil {
+				// If the dependencies failed to delete because of errors, retry again
+				return reconcile.Result{}, err
+			}
+			if isDeleted {
+				reqLogger.Info("Removing finalizer")
+				// Once all finalizers have been removed, the object will be deleted
+				sc.ObjectMeta.Finalizers = remove(sc.ObjectMeta.Finalizers, storageClusterFinalizer)
+				if err := r.client.Update(context.TODO(), sc); err != nil {
+					reqLogger.Error(err, "Failed to remove finalizer from storagecluster")
+					return reconcile.Result{}, err
+				}
+			} else {
+				// Watch resources and events and reconcile.
+				return reconcile.Result{}, nil
+			}
+		}
+		reqLogger.Info("Object is terminated, skipping reconciliation")
+		return reconcile.Result{}, nil
+	}
+
+	// in-memory conditions should start off empty. It will only ever hold
+	// negative conditions (!Available, Degraded, Progressing)
+	r.conditions = nil
+	// Start with empty r.phase
+	r.phase = ""
+
+	for _, f := range []func(*ocsv1.StorageCluster, logr.Logger) error{
+		// Add support for additional resources here
+		r.ensureCephConfig,
+		r.ensureExternalCephCluster,
+		r.ensureNoobaaSystem,
```

Comment:
it would be better if we comment it out here in this PR and send a separate PR for noobaa work.

Reply:
Done...